src/nunit.analyzers/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressor.cs

```diff
@@ -61,38 +61,116 @@ private static bool ShouldBeSuppressed(SyntaxNode node)
+                        (block is not null && IsValidatedNotNullByPreviousStatementInSameBlock(possibleNullReference, block, statement)))
                     {
                         return true;
                     }
                 }
 
-                currentNode = parent;
+                currentNode = statement.Parent;
             }
 
             return false;
         }
 
-        private static bool IsValidatedNotNullByPreviousStatementInSameBlock(string possibleNullReference, BlockSyntax parent, StatementSyntax statement)
+        /// <summary>
+        /// Determine which side of the conditional expression if failing nullability.
+        /// </summary>
+        /// <remarks>
+        /// We only try to detect simple cases where the operands being tested for <see langword="null"/> is one of the operands being returned:
+        /// <code>variable is not null ? variable : otherExpression</code>
+        /// </para>
+        /// We recognize the 'is' pattern operations and normal equality.
+        /// </remarks>
+        /// <param name="conditionalExpression">Conditional expression to investigate.</param>
+        /// <returns>Either the 'WhenTrue' or the 'WhenFalse' part of the <paramref name="conditionalExpression"/>.</returns>
+        private static ExpressionSyntax? DetermineConditionalExpressionPath(ConditionalExpressionSyntax conditionalExpression)
+        {
+            string? testedAgainstNullExpression;
+            bool isNot = false;
+
+            if (conditionalExpression.Condition is IsPatternExpressionSyntax patternExpression &&
+                patternExpression.Expression is IdentifierNameSyntax or MemberAccessExpressionSyntax)
+            {
+                // identifier is
+                testedAgainstNullExpression = patternExpression.Expression.ToString();
+                PatternSyntax pattern = patternExpression.Pattern;
+                if (pattern is UnaryPatternSyntax unaryPattern && unaryPattern.IsKind(SyntaxKind.NotPattern))
+                {
+                    // 'expression' is not
+                    isNot = true;
+                    pattern = unaryPattern.Pattern;
+                }
+
+                if (pattern is not ConstantPatternSyntax constantPattern || !constantPattern.Expression.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    return null;
+                }
+
+                // identifier is (not) null
+            }
+            else if (conditionalExpression.Condition is BinaryExpressionSyntax binaryExpression &&
+                binaryExpression.Right.IsKind(SyntaxKind.NullLiteralExpression) &&
+                binaryExpression.Left is IdentifierNameSyntax or MemberAccessExpressionSyntax)
+            {
+                testedAgainstNullExpression = binaryExpression.Left.ToString();
+                isNot = binaryExpression.IsKind(SyntaxKind.NotEqualsExpression);
+            }
+            else
```

Comment:
Should we also cater for the people having a C/C++ background of hence often place the `null` to the left - e.g. `string fatal = null != line2 ? line2 : line1;`. Although it is has been some time since I've seen this formatting used in C#

Reply:
If they would have StyleCop enabled they would have been told SA1131: Constant values should appear on the right-hand side of comparisons. But yes, I have seen code like that :-( Added branch.

Reply:
My mistake I meant `block` and not `parent` in the question. Thanks for the thorough explanation, as far as I can tell it makes sense.